invenio_communities/assets/semantic-ui/js/invenio_communities/community/frontpage.js

```diff
@@ -1,175 +1,19 @@
-};
-
-class CommunityCard extends Component {
-  render() {
-    const { community, defaultLogo } = this.props;
-    return (
-      <Card fluid href={`/communities/${community.slug}`}>
-        <Image
-          wrapped
-          centered
-          ui={false}
-          src={community.links.logo}
-          fallbackSrc={defaultLogo}
-          loadFallbackFirst
-        />
-        <Card.Content>
-          <Card.Header>
-            {_truncate(community.metadata.title, { length: 30 })}
-          </Card.Header>
-          {community.metadata.description && (
-            <Card.Description>
-              <div className="truncate-lines-2">{community.metadata.description}</div>
-            </Card.Description>
-          )}
-        </Card.Content>
-      </Card>
-    );
-  }
-}
-
-CommunityCard.propTypes = {
-  community: PropTypes.object.isRequired,
-  defaultLogo: PropTypes.string.isRequired,
-};
-
-class CommunitiesCardGroup extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isLoading: false,
-      data: { hits: [] },
-    };
-  }
-
-  componentDidMount() {
-    this.fetchData();
-  }
-
-  componentWillUnmount() {
-    this.cancellableFetch && this.cancellableFetch.cancel();
-  }
-
-  fetchData = async () => {
-    const { fetchDataUrl } = this.props;
-    this.setState({ isLoading: true });
-    this.cancellableFetch = withCancel(http.get(fetchDataUrl));
-
-    try {
-      const response = await this.cancellableFetch.promise;
-
-      this.setState({ data: response.data.hits, isLoading: false });
-    } catch (error) {
-      // TODO: handle error response
-    }
-  };
-
-  renderCards() {
-    const { data } = this.state;
-    const { defaultLogo } = this.props;
-
-    return data.hits.map((community) => {
-      return (
-        <CommunityCard
-          key={community.id}
-          community={community}
-          defaultLogo={defaultLogo}
-        />
-      );
-    });
-  }
-
-  render() {
-    const { isLoading, data } = this.state;
-    const { emptyMessage } = this.props;
-    return (
-      <PlaceholderLoader isLoading={isLoading}>
-        {data.hits.length === 0 ? (
-          <EmptyMessage message={emptyMessage} />
-        ) : (
-          <Card.Group
-            doubling
-            stackable
-            itemsPerRow={5}
-            className="community-frontpage-cards"
-          >
-            {this.renderCards()}
-          </Card.Group>
-        )}
-      </PlaceholderLoader>
-    );
-  }
-}
-
-CommunitiesCardGroup.propTypes = {
-  fetchDataUrl: PropTypes.string.isRequired,
-  defaultLogo: PropTypes.string.isRequired,
-  emptyMessage: PropTypes.string.isRequired,
-};
+import CommunitiesCardGroup from "./CommunitiesCardGroup";
 
 const userCommunitiesContainer = document.getElementById("user-communities");
-const featuredCommunitiesContainer = document.getElementById("featured-communities");
+const newCommunitiesContainer = document.getElementById("new-communities");
```

Comment:
This might be a breaking change, if someone is overriding the Jinja template. WDYT?
We need to document it.

Reply:
Good point. I've added a warning about it in the CHANGES.rst . I am thinking that when v12 is released, the release page should have a section pointing to the CHANGES.rst of the modules. Manually listing all the changes would be ideal but seems too time consuming.